packages/eslint-plugin/src/util/isNodeEqual.ts

```diff
@@ -0,0 +1,47 @@
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/utils';
+/*
+ * Supported:
+ *   Identifier Literal MemberExpression ThisExpression
+ *   TSNamedTupleMember TSNamespaceExportDeclaration TSNeverKeyword
+ *   TSNonNullExpression TSNullKeyword TSNumberKeyword TSObjectKeyword
+ *   TSOptionalType TSParameterProperty TSPrivateKeyword TSPropertySignature
+ *   TSProtectedKeyword TSPublicKeyword TSQualifiedName TSReadonlyKeyword
+ *   TSRestType TSStaticKeyword TSStringKeyword TSSymbolKeyword
+ *   TSTemplateLiteralType TSThisType TSTupleType TSTypeAliasDeclaration
+ *   TSTypeAnnotation TSTypeAssertion TSTypeLiteral TSTypeOperator
+ *   TSTypeParameter TSTypeParameterDeclaration TSTypeParameterInstantiation
+ *   TSTypePredicate TSTypeQuery TSTypeReference TSUndefinedKeyword TSUnionType
+ *   TSUnknownKeyword TSVoidKeyword UnaryExpression UpdateExpression
+ *   VariableDeclaration VariableDeclarator WhileStatement WithStatement
+ *   YieldExpression
+ */
```

Comment:
Coming back to this (https://github.com/typescript-eslint/typescript-eslint/pull/4965#discussion_r894990290), I don't think we need this _Supported_ list either to be honest. There's not much use in having a list of several dozen node types. Folks aren't likely to read through it, and there's nothing enforcing it does work beyond the unit tests. Which are documentation enough themselves.

```suggestion
```